connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/ErrorFactoryBuilder.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connect.client
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+private[client] class ErrorFactoryBuilder() {
+
+  private val constructors = mutable.Map[String, (String, Throwable) => Throwable]()
+
+  def addConstructor[T <: Throwable: ClassTag](
+      throwableCtr: (String, Throwable) => T): ErrorFactoryBuilder = {
```

Comment:
`(String, Throwable) => T`, what is `Throwable` for? Would be great if we can have some doc.

Reply:
Throwable is cause. I added documentation.